src/renderer/views/stake/StakeView.tsx

```diff
@@ -12,28 +14,44 @@ import { useParams } from 'react-router-dom'
 import { AddStakeStory } from '../../components/stake/AddStake/AddStake.stories'
 import { Stake } from '../../components/stake/Stake'
 import BackLink from '../../components/uielements/backLink'
-/**
- * temporary data mock
- * @TODO @thatStrangeGuyThorchain replace mocck after
- * https://github.com/thorchain/asgardex-electron/issues/441 resolved
- */
-import { defaultPoolShare } from '../../components/uielements/poolShare/PoolShare.stories'
+import { useBinanceContext } from '../../contexts/BinanceContext'
 import { useMidgardContext } from '../../contexts/MidgardContext'
+import { useWalletContext } from '../../contexts/WalletContext'
 import { StakeRouteParams } from '../../routes/stake'
 import { PoolDetailsView } from './PoolDetails/PoolDetailsView'
+import { ShareView } from './Share/ShareView'
 
 type Props = {}
 
 const StakeView: React.FC<Props> = (_) => {
   const { asset: assetParam } = useParams<StakeRouteParams>()
   const { service: midgardService } = useMidgardContext()
+  const { keystoreService } = useWalletContext()
+  const { address$ } = useBinanceContext()
```

Comment:
Since we will provide other chains than Binance in the near future, address should be depending on a `chain`. The current `chain` (or better: the chain the current asset runs on) can we get from `Asset`. With this information we could create a helper `addressByChain(chain: Chain): O.Option<string>` (or similar), which provides such address. What do you think?

Reply:
yep. i thought about it at the beginning but forgot at the end :))

Reply:
Feel free to add this with another PR. Could you create an issue for it before, pls.